src/Lookup/InProcessCachingDataTypeLookup.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Wikibase\DataModel\Services\Lookup;
+
+use Wikibase\DataModel\Entity\PropertyId;
+
+/**
+ * PropertyDataTypeLookup that provides in-process caching.
+ *
+ * @since 3.1
+ *
+ * @licence GNU GPL v2+
+ * @author Katie Filbert < aude.wiki@gmail.com >
+ */
+class InProcessCachingDataTypeLookup implements PropertyDataTypeLookup {
```

Comment:
Shows I did not get enough sleep yesterday :)

Having a CachingPropertyDataTypeLookup that takes a caching interface (cannot be HashBagOStuff) would be nice, though I think having a specific decorator as done here is also fine.